File: Java/app/src/main/java/software/rsquared/template/di/ViewModelModule.java
package software.rsquared.template.di;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;

import dagger.Binds;
import dagger.Module;
import dagger.multibindings.IntoMap;
import software.rsquared.template.ui.simple.SampleFragment.SampleViewModel;
import software.rsquared.template.ui.pager.PagerViewModel;

/**
 * @author rSquared.software
 */
@Module
abstract class ViewModelModule {

    @Binds
    @IntoMap
    @ViewModelKey(PagerViewModel.class)
    abstract ViewModel bindPagerViewModel(PagerViewModel pagerViewModel);

    @Binds
    @IntoMap
    @ViewModelKey(SampleViewModel.class)
    abstract ViewModel bindSampleViewModel(SampleViewModel sampleViewModel);

    @Binds
    abstract ViewModelProvider.Factory bindViewModelFactory(ViewModelFactory factory);
}

File: app/src/main/java/software/rsquared/template/di/ViewModelModule.kt
package software.rsquared.template.di

import androidx.lifecycle.ViewModelProvider
import dagger.Binds
import dagger.Module

/**
 * @author Rafal Orlik
 */
@Module
abstract class ViewModelModule {

	@Binds
	internal abstract fun bindViewModelFactory(factory: ViewModelFactory): ViewModelProvider.Factory

}
